### PR TITLE
Remove support for `exportsNamespace` and get rid of `envInfo`.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

File: test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala [hunk 1]
```diff
@@ -25,8 +25,24 @@ import org.scalajs.testsuite.utils.AssertThrows.assertThrows
...
+   *  module-global variable, which we can retrieve as if it were in the global
+   *  scope.
+   */
+  val exportsNamespace: js.Dynamic = {
+    if (Platform.isNoModule)
+      org.scalajs.testsuite.utils.JSUtils.globalObject
+    else if (Platform.isCommonJSModule)
+      js.Dynamic.global.exports
+    else
+      ???
```
Comment (gzm0) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala]: Maybe add something more descriptive?

File: library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala [hunk 2]
```diff
@@ -1,46 +0,0 @@
...
- *  defined in its scope (and references an object). If so, it uses it as
- *  environment info.
- *  Missing, non-optional fields (according to this facade type) are initialized
- *  to default values, optional fields are kept as in the original object.
- *  Finally, [[js.Object.freeze]] is called on the object to avoid modification.
- *
- *  @groupname envInfo Scala.js environment configuration
- *  @groupprio envInfo 1
- */
-sealed trait EnvironmentInfo extends js.Object {
```
Comment (gzm0) [inline library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala]: I do not understand why you are getting rid of the typing here. It seems like this leads to very un-idiomatic Scala.js.

How about:

```
sealed trait EnviromentInfo // unchanged

object EnvironmentInfo {
  def envInfo: js.UndefOr[EnvironmentInfo] = {
    // ...
  }
}
```

File: test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala (hunk 1 above)
Comment (sjrd) [inline test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala]: I thought that `NotImplementedError` was the best possible description in this case. I'll write a proper error message.

File: library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala (hunk 2 above)
Comment (sjrd) [inline library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala]: Er, yes, good point. <insert shame emoji here>

**Comment by sjrd** [comment]
Updated.

File: library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala [hunk 3]
```diff
@@ -50,3 +41,18 @@ sealed trait EnvironmentInfo extends js.Object {
...
 }
+
+object EnvironmentInfo {
+  /** The value of the `__ScalaJSEnv` variable, if it exists. */
+  val envInfo: js.UndefOr[EnvironmentInfo] = {
+    import js.Dynamic.{global => g}
+
+    // We've got to use selectDynamic explicitly not to crash Scala 2.10
+    if (js.typeOf(g.selectDynamic("__ScalaJSEnv")) == "object" &&
+        js.DynamicImplicits.truthValue(g.selectDynamic("__ScalaJSEnv"))) {
```
Comment (gzm0) [inline library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala]: When is this line needed?

Comment (gzm0) [inline library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala]: `null` I guess? Consider simply comparing to `null` for improved clarity.

Comment (sjrd) [inline library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala]: Indeed. It was a direct translation of the JS code and I haven't given it any thought. Comparing to `null` would be better indeed.

**Pull request merged**